tests/testthat/test-timeseries_plot.R

```diff
@@ -1,40 +1,106 @@
-# mydata$station_var <- ifelse(mydata$ws < 5, "A", "B")
-# 
-# # Test the data -------------------------------------------------------------------------------------------------------
-# # no station
-# timeseries_plot(mydata, date_time = "date", elements = c("nox"), facets = "none")
-# timeseries_plot(mydata, date_time = "date", elements = c("nox"), facets = "elements")
-# timeseries_plot(mydata, date_time = "date", elements = c("nox"), facets = "stations")
-# timeseries_plot(mydata, date_time = "date", elements = c("nox"), facets = "both")
-# 
-# # station
-# timeseries_plot(mydata, date_time = "date", elements = c("nox"), station = "station_var", facets = "none")
-# timeseries_plot(mydata, date_time = "date", elements = c("nox"), station = "station_var", facets = "elements")
-# timeseries_plot(mydata, date_time = "date", elements = c("nox"), station = "station_var")
-# timeseries_plot(mydata, date_time = "date", elements = c("nox"), station = "station_var", facets = "both")
-# 
-# # no station
-# timeseries_plot(mydata, date_time = "date", elements = c("nox", "no2"), facets = "none", show.legend = FALSE) +
-#   viridis::scale_color_viridis(discrete = TRUE, option = "C")
-# timeseries_plot(mydata, date_time = "date", elements = c("nox", "no2"), facets = "elements")
-# 
-# # station
-# timeseries_plot(mydata, date_time = "date", elements = c("nox", "no2"), station = "station_var", facets = "none")
-# timeseries_plot(mydata, date_time = "date", elements = c("nox", "no2"), station = "station_var", facets = "elements")
-# timeseries_plot(mydata, date_time = "date", elements = c("nox", "no2"), station = "station_var", facets = "stations")
-# timeseries_plot(mydata, date_time = "date", elements = c("nox", "no2"), station = "station_var", facets = "both")
-# 
-# 
-# 
-# # Should throw error
-# #timeseries_plot("mydata", date_time = "date", elements = c("nox", "max_tmax"), station = "station_var", facets = "none")
-# #timeseries_plot(mydata, date_time = year, elements = c("nox", "max_tmax"), station = "station_var", facets = "none")
-# #timeseries_plot(mydata, date_time = "date", elements = c("no"), station = "station_var", facets = "none")
-# #timeseries_plot(mydata, date_time = "date", elements = c("nox"), facets = "none")
-# #timeseries_plot(mydata, date_time = "date", elements = c("nox"), facets = "nonxe")
-# #timeseries_plot(mydata, date_time = "date", elements = c("nox"))
+library(cdms.products)
+library(dplyr)
+library(ggplot2)
+library(tidyr)
+
+niger50 <- daily_niger %>%
+  dplyr::filter(year == 1950)
+
+agades <- niger50 %>%
+  dplyr::filter(station_name == "Agades")
+
+test_that("single element facet by station graphs are correct", {
+  t1 <- timeseries_plot(data = niger50, date_time = "date", elements = "tmax", 
+                        station = "station_name", facets = "stations")
+  r1 <- ggplot(niger50, aes(x = date, y = tmax)) +
+    geom_line() +
+    facet_wrap(vars(station_name))
+  expect_equal(gg_data(t1), gg_data(r1))
```

Comment:
@lilyclements @HawardKetoyoMsatsi See here I have created a function `gg_data` which extracts the data that's plotted in a gg graph. So it's a good way to roughly check they are equal - on variables, layers, aesthetics and facets. It won't check theme aspects, we could think of adding functions to do this later.